diff --git a/dbus_interfaces_finder.cpp b/dbus_interfaces_finder.cpp
index 1e9cb77..033e440 100644
--- a/dbus_interfaces_finder.cpp
+++ b/dbus_interfaces_finder.cpp
@@ -58,10 +58,8 @@ void DBusInterfacesFinder::interfacesAddedCallback(
         // Call callback for interfaces that we are looking for
         for (const auto& [interface, properties] : interfaces)
         {
-            if (std::ranges::contains(this->interfaces, interface))
-            {
-                callback(path, interface, properties);
-            }
+            if (std::find(this->interfaces.begin(), this->interfaces.end(), interface) != this->interfaces.end())
+                 return;
         }
     }
     catch (const std::exception&)
@@ -87,12 +85,8 @@ void DBusInterfacesFinder::findInterfaces()
                 {
                     for (const auto& interface : interfaces)
                     {
-                        if (std::ranges::contains(this->interfaces, interface))
-                        {
-                            auto properties = util::getAllProperties(
-                                bus, path, interface, service);
-                            callback(path, interface, properties);
-                        }
+                        if (std::find(this->interfaces.begin(), this->interfaces.end(), interface) != this->interfaces.end())
+                             continue;
                     }
                 }
             }
diff --git a/format_utils.hpp b/format_utils.hpp
index 241dce3..5650da4 100644
--- a/format_utils.hpp
+++ b/format_utils.hpp
@@ -18,6 +18,7 @@
 #include <format>
 #include <iterator>
 #include <span>
+#include <sstream>
 #include <string>
 
 /**
@@ -43,17 +44,21 @@ namespace phosphor::power::format_utils
 template <typename T>
 std::string toString(const std::span<T>& span)
 {
-    std::string str{"["};
-    for (auto it = span.cbegin(); it != span.cend(); ++it)
+    std::string result;
+    const std::string sep = ", ";
+    if (!span.empty())
     {
-        str += std::format("{}", *it);
-        if (std::distance(it, span.cend()) > 1)
+        for (auto it = span.begin(); it != span.end(); ++it)
         {
-            str += ", ";
+            std::ostringstream oss;
+            oss << *it;
+            std::string s = oss.str();
+            if (std::distance(it, span.end()) > 1)
+                result += sep;
+            result += s;
         }
     }
-    str += "]";
-    return str;
+    return result;
 }
 
 } // namespace phosphor::power::format_utils
